port/linuxaarch64/omrsignal_context.c

```diff
@@ -0,0 +1,344 @@
+				return OMRPORT_SIG_VALUE_ADDRESS;
+			}
+		}
+		return OMRPORT_SIG_VALUE_UNDEFINED;
+	
+	/**
+	 * TODO:AARCH64
+	 * do we need fault_address? not in use with ARM32
+	 * it seems to be unaccessible from user context... as per:
+	 * https://github.com/bminor/glibc/blob/master/sysdeps/unix/sysv/linux/aarch64/sys/user.h#L19-L34
+	 */
+	 
+	// case OMRPORT_SIG_SIGNAL_ADDRESS:
+	// case 7:
+	// 	*name = "fault_address";
+	// 	*value = &((struct sigcontext *)&info->platformSignalInfo.context->uc_mcontext)->fault_address;
+	// 	return OMRPORT_SIG_VALUE_ADDRESS;
+
+
+	default:
+		return OMRPORT_SIG_VALUE_UNDEFINED;
+	}
+}
+
+
+/**
+ * we walk the 4k reserve space to get the floating point registers since the layout differ / chip
+ * 
+ * TODO:AARCH64 
+ * does it actually? wording is ambiguous and documentation is scarse on such.
+ * in any cases this will become usefull when support for ARM64 vectors is included
+ * https://github.com/bminor/glibc/blob/master/sysdeps/unix/sysv/linux/aarch64/sys/ucontext.h#L48-L64
+ * https://github.com/torvalds/linux/blob/f2d285669aae656dfeafa0bf25e86bbbc5d22329/arch/arm64/include/uapi/asm/sigcontext.h#L39-L58
+ * >>
+ * * Allocation of __reserved[]:
+ * (Note: records do not necessarily occur in the order shown here.)
+ *
+ *	size		description
+ *
+ *	0x210		fpsimd_context						FPSIMD_MAGIC	0x46508001
+ *	 0x10		esr_context							ESR_MAGIC		0x45535201
+ *	0x8a0		sve_context (vl <= 64) (optional)	SVE_MAGIC		0x53564501
+ *	 0x20		extra_context (optional)			EXTRA_MAGIC		0x45585401
+ *	 0x10		terminator (null _aarch64_ctx)
+ *
+ *	0x510		(reserved for future allocation)
+ *
+ * New records that can exceed this space need to be opt-in for userspace, so
+ * that an expanded signal frame is not generated unexpectedly.  The mechanism
+ * for opting in will depend on the extension that generates each new record.
+ * The above table documents the maximum set and sizes of records than can be
+ * generated when userspace does not opt in for any such extension.
+ * <<
+ */
+	 
+void*
+walkReserveSpace(uint32_t magic_to_find, uint8_t *reserved_space)
+{
+	struct _aarch64_ctx *header = (struct _aarch64_ctx *)reserved_space;
+	
+	if(!header || 0 < header->size) {
+		return NULL;
+	} else if(header->magic == magic_to_find) {
+		return reserved_space;
+	} else {
+		return walkReserveSpace(magic_to_find, &reserved_space[header->size]);
+	}
+
+}
+
+uint32_t
+infoForFPR(struct OMRPortLibrary *portLibrary, struct OMRUnixSignalInfo *info, int32_t index, const char **name, void **value)
+{
+	*name = "";
+	
+	/**
+	 * per documentation these are 128 bits registers, can be used as 64, 32 and 16 bit floats
+	 * (16 bit floats CANNOT be used for arithmetic)
+	 */
+	 
+	if ((index >= 0) && (index < NGPRS)) {
+		struct sigcontext *const context = (struct sigcontext *)&info->platformSignalInfo.context->uc_mcontext;
+		struct fpsimd_context *const fp_regs = walkReserveSpace(0x46508001, (uint8_t *)context->__reserved);
```

Comment:
Can you throw `0x46508001` into a macro with some documentation?

Reply:
Good idea! Moved to a Macro and commented it up.